### PR TITLE
HSDS Depth

### PR DESCRIPTION
**Pull request by tinkertrain**
Screen Shot 2020-10-07 at 1 40 27 PM

We use different levels of depth across our components but currently they're all coded into each component that needs them, this has lead to some inconsistencies and repetition, with this PR we aim to eliminate those issues and make it easier to incorporate Depth into our components with the assurance that by using the provided tools you'll always be following the HSDS style guide.

> Based on the HSDS Depth Guide https://cdpn.io/team/helpscout/debug/jOWBppq

### 👀 Check it out in storybook

- **[Depth guide](https://deploy-preview-888--hsds-react.netlify.app/?path=/story/style-2-depth--page)**
- **[Component](https://deploy-preview-888--hsds-react.netlify.app/?path=/docs/utilities-ui-convenience-depth--default-story#depth)**
- **[Mixins](https://deploy-preview-888--hsds-react.netlify.app/?path=/docs/utilities-ui-convenience-depth--default-story#usage-as-css-mixins)**

This PR includes:

- The aforementioned depth guide inside the style section in storybook
- `<Depth />` component
- Depth levels as mixins
- Detailed documentation
- Moves the "Guides" into their own folder in the repo for better organization

### CODE REVIEW

File: src/components/Depth/Depth.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import { Depth } from '@helpscout/hsds-react'
+
+/**
+ * You can set any <Depth> prop here,
+ * if you need this container to be flexible and be able to provide
+ * custom props value to that particular prop,
+ * add something like `level: props.level || 300`
+ */
+export const LabelWithDepth = styled(Depth).attrs(props => ({
+  level: 300,
```
Comment (plbabin) [inline src/components/Depth/Depth.stories.mdx:82]: that's a clever way to set props to an extendable styled components 🎉

File: src/components/Depth/Depth.storiesHelpers.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  font-size: 14px;
+  color: #1292ee;
+
+  &:hover {
+    color: #0077cc;
+    background-color: #d6edff;
+    text-decoration: underline;
+  }
+
+  &.figma {
```
Comment (plbabin) [inline src/components/Depth/Depth.storiesHelpers.js:135]: I really like the related new section.

Figma link icon are not working on the netlify preview, not sure why

Image 2020-10-07 at 1 40 07 PM

**Review by plbabin — APPROVED** [review]
looks good, nice work with the documentation!

**Review by digitaldesigner — APPROVED** [review]
I can't express in words how happy this makes me!

File: src/components/Depth/Depth.storiesHelpers.js (hunk 2 above)
Comment (tinkertrain) [reply src/components/Depth/Depth.storiesHelpers.js:135]: Thanks, fixed!

**Pull request merged**